### DISCUSSION TITLE
如何快速编译文档

### DISCUSSION THREAD

**Discussion by MA-Wenbo** [sjtug/SJTUThesis#889] [展示区]
该如何不包含文件头尾，快速编译文档呢？

或者，能不能只编译某个章节呢？

或者，有没有交大论文模板，环境、正文相同，但是不包含那么多的版面内容呢

```
$ make all
'xelatex': time = 4.35
'xelatex': time = 4.40
'xelatex': time = 4.27
'xdvipdfmx': time = 0.96
Accumulated processing time = 14.91
Number of rules run = 4
```

**Comment by LogCreative** [comment]
最简单的方法就是使用 `%` 注释掉对应的部分。

当然，如果你不想影响章节间的交叉引用，可以使用 `\include` 与 `\includeonly`，见 [LearnLaTeX - 组建更长的文档](https://github.com/CTeX-org/learnlatex.github.io/blob/zh-Hans/zh-Hans/lesson-13.md)。

**Reply by MA-Wenbo** [reply]
我注释掉了所有不需要的部分，现在只有目录、摘要和正文。目前需要七秒钟编译。

我还按照这里的方法注释掉了tikz
https://github.com/sjtug/SJTUThesis/discussions/828#discussioncomment-3056016

这句话我也尝试了，但是会报错
```
\documentclass[type=draft]{sjtuthesis} % very quick
```

现在需要7秒钟编译。

请问可以再快一些吗？

**Reply by LogCreative** [reply]
应该是

```latex
\documentclass[type=master,draft]{sjtuthesis}
```

即使不用 `draft`，也应该使用合适大小的图片（必要时压缩一下）。

**Reply by MA-Wenbo** [reply]
这句话成功使用了，似乎有10%的提升。请问还有没有什么可以丢弃的package呢？

**Reply by LogCreative** [reply]
应该没有更多的 package 需要被丢弃了，它们不会拖慢很多速度。如果你实在想从头构建环境，推荐阅读 [SJTUThesis 文档](https://github.com/sjtug/SJTUTeX/releases/latest)。
